scripts/test-coverage-upload.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+set -o pipefail
+
+LOG_FILE=${1:-test-coverage.log}
+COVERAGE_DIR=${COVERAGE_DIR:-coverage_dir}
+COVERAGE_FILE="${COVERAGE_DIR}/coverage.txt"
+
+mkdir -p "${COVERAGE_DIR}"
+
+echo "=== Running tests and generating coverage report ==="
+go test -coverprofile="${COVERAGE_FILE}" ./... 2>&1 | tee "${LOG_FILE}"
```

Comment:
we don't need the script file, just need to add these cover params to test-unit.sh.